### PR TITLE
SLM-162 UI updates

### PR DESCRIPTION
**Pull request by mikehalmamoj**
Various changes following a design review

### CODE REVIEW

File: assets/sass/local.sass [hunk 1]
```diff
@@ -1,6 +1,3 @@
-.govuk-main-wrapper
-  min-height: 600px
-
```
Comment (mikehalmamoj) [inline assets/sass/local.sass:3]: SLM-162 AC4

File: integration_tests/pages/barcode/reviewRecipients.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ export default class ReviewRecipientsPage extends Page {
     // Get row `recipientNumber` from the table, then get its 5th td cell to click the link within it
     this.recipientsTableBodyRows()
       .eq(recipientNumber - 1)
-      .find(`td:nth-of-type(5) a`)
+      .find(`td:nth-of-type(4) a`)
```
Comment (mikehalmamoj) [inline integration_tests/pages/barcode/reviewRecipients.ts:27]: A few fixes after changing column 1 from rd to th: SLM-164 AC8

File: server/views/pages/barcode/create-new-contact-by-prisoner-name.njk [hunk 3]
```diff
@@ -49,16 +49,19 @@
                 {
                   name: "day",
                   classes: "govuk-input--width-2" + dobErrorClass,
+                  attributes: { 'maxlength': 2 },
```
Comment (mikehalmamoj) [inline server/views/pages/barcode/create-new-contact-by-prisoner-name.njk:52]: SLM-164 AC12

File: server/views/pages/barcode/generate-barcode-image.njk [hunk 4]
```diff
@@ -20,10 +20,10 @@
     {% endif %}
 
     <div class="govuk-grid-row govuk-!-margin-bottom-4">
-      <div class="govuk-grid-column-full">
+      <div class="govuk-grid-column-three-quarters">
```
Comment (mikehalmamoj) [inline server/views/pages/barcode/generate-barcode-image.njk:23]: SLM-164 AC6

File: server/views/pages/barcode/generate-barcode-image.njk [hunk 5]
```diff
@@ -34,7 +34,7 @@
     </div>
 
     {% for barcodeImage in barcodeImages %}
-      <div class="govuk-grid-row govuk-!-margin-bottom-6">
+      <div class="govuk-grid-row">
```
Comment (mikehalmamoj) [inline server/views/pages/barcode/generate-barcode-image.njk:37]: SLM-164 AC10

File: server/views/pages/barcode/review-recipients.njk [hunk 6]
```diff
@@ -23,7 +23,7 @@
 
         {% if recipients.length > 0 %}
           {{ govukTable({
-            firstCellIsHeader: false,
+            firstCellIsHeader: true,
```
Comment (mikehalmamoj) [inline server/views/pages/barcode/review-recipients.njk:26]: SLM-164 AC8

File: server/views/pages/barcode/review-recipients.njk [hunk 7]
```diff
@@ -36,7 +36,7 @@
           }) }}
 
           <p class="govuk-!-margin-top-6 govuk-!-margin-bottom-6">
-            <a href="/barcode/find-recipient" class="govuk-link" data-qa="add-another-recipient">Add another recipient</a>
+            <a href="/barcode/find-recipient" class="govuk-link govuk-link--no-visited-state" data-qa="add-another-recipient">Add another recipient</a>
```
Comment (mikehalmamoj) [inline server/views/pages/barcode/review-recipients.njk:39]: SLM-164 AC9

File: server/views/pages/link/emailSent.njk [hunk 8]
```diff
@@ -11,13 +11,13 @@
 
   <section class="app-container govuk-body">
     <div class="govuk-grid-row">
-      <div class="govuk-grid-column-full">
+      <div class="govuk-grid-column-two-thirds">
```
Comment (mikehalmamoj) [inline server/views/pages/link/emailSent.njk:14]: SLM-164 AC1

File: server/views/pages/link/requestLink.njk [hunk 9]
```diff
@@ -19,10 +19,9 @@
     {% endif %}
 
     <div class="govuk-grid-row">
-      <div class="govuk-grid-column-full">
+      <div class="govuk-grid-column-two-thirds">
```
Comment (mikehalmamoj) [inline server/views/pages/link/requestLink.njk:22]: SLM-164 AC1

File: server/views/pages/link/requestLink.njk [hunk 10]
```diff
@@ -19,10 +19,9 @@
...
         <h1 class="govuk-heading-xl">Request a link to sign in</h1>
-        <h2 class="govuk-heading-s">Enter your Criminal Justice Secure Mail (CJSM) email address.</h2>
-        <p>You'll receive an email containing a link to the service in your inbox. Follow the link to sign in.</p>
+        <p>Enter your Criminal Justice Secure Mail (CJSM) email. You'll receive an email containing a link to the service in your inbox. Follow the link to sign in.</p>
```
Comment (mikehalmamoj) [inline server/views/pages/link/requestLink.njk:24]: SLM-164 AC2

File: server/views/pages/link/requestLink.njk [hunk 11]
```diff
@@ -35,12 +34,18 @@
...
+              },
+              hint: {
+                text: 'For example, name@myorganisation.com.cjsm.net'
               },
               id: "email",
               name: "email",
               value: form.email,
               type: 'email',
+              spellcheck: false,
+              autocomplete: 'email',
```
Comment (mikehalmamoj) [inline server/views/pages/link/requestLink.njk:48]: SLM-164 AC3

File: server/views/pages/barcode/edit-contact-details.njk [hunk 12]
```diff
@@ -78,6 +78,7 @@
               id: "prisonNumber",
               name: "prisonNumber",
               value: form.prisonNumber,
+              classes: 'govuk-input--width-10',
```
Comment (mikehalmamoj) [inline server/views/pages/barcode/edit-contact-details.njk:81]: SLM-164 AC11

File: server/filters/recipientTableRowsFilter.test.ts [hunk 13]
```diff
@@ -34,15 +34,27 @@ describe('recipientTableRowsFilter', () => {
         { text: 'John Smith' },
         { text: 'A1234BC' },
         { text: 'HMP BRINSFORD' },
-        { html: '<a href="/barcode/edit-contact/1" class="govuk-link">Edit details</a>' },
-        { html: '<a href="/barcode/review-recipients/remove/0" class="govuk-link">Remove</a>' },
+        {
+          html: '<a href="/barcode/edit-contact/1" class="govuk-link govuk-link--no-visited-state">Edit details</a>',
+          classes: 'govuk-table__cell--numeric',
```
[the remark this answers is not in the document]
Comment (nathanrussell-moj-digital) [reply server/filters/recipientTableRowsFilter.test.ts:39]: Table cell numeric? I can see how that would achieve the desired outcome, but it doesnt feel semantically correct?
Did Richard point you at that class? (he seems pretty hot on knowing which classes to use etc)

Would `govuk-!-text-align-right` on the anchor link be a better fit? (I dont know if that would work, but it reads better to me?)

[the remark this answers is not in the document]
Comment (mikehalmamoj) [reply server/filters/recipientTableRowsFilter.test.ts:39]: It doesn't work if you apply it to the anchor, it has to be applied to the table cell. Yes Richard suggested the class. I looked into a sass rule that could be applied to the anchor but apparently CSS still doesn't support selecting a parent - could have started down the road of some complicated sass function but it doesn't seem worth it. So I just went with Richard's suggestion.

[the remark this answers is not in the document]
Comment (nathanrussell-moj-digital) [reply server/filters/recipientTableRowsFilter.test.ts:39]: Perfect, no probs. Just thought I'd ask 👍

**Review by nathanrussell-moj-digital — APPROVED** [review]

**Pull request merged**